allennlp/training/metrics.py

```diff
@@ -112,6 +123,73 @@ def reset(self):
         self.total_count = 0.0
 
 
+@Metric.register("boolean_accuracy")
+class BooleanAccuracy(Metric):
+    """
+    Just checks batch-equality of two tensors and computes an accuracy metric based on that.  This
+    is similar to :class:`CategoricalAccuracy`, if you've already done a ``.max()`` on your
+    predictions.  If you have categorical output, though, you should typically just use
+    :class:`CategoricalAccuracy`.  The reason you might want to use this instead is if you've done
+    some kind of constrained inference and don't have a prediction tensor that matches the API of
+    :class:`CategoricalAccuracy`, which assumes a final dimension of size ``num_classes``.
+    """
+    def __init__(self) -> None:
+        self._correct_count = 0.
+        self._total_count = 0.
+
+    def __call__(self,
+                 predictions: torch.Tensor,
+                 gold_labels: torch.Tensor,
+                 mask: Optional[torch.Tensor] = None):
+        """
+        Parameters
+        ----------
+        predictions : ``torch.Tensor``, required.
+            A tensor of predictions of shape (batch_size, ...).
+        gold_labels : ``torch.Tensor``, required.
+            A tensor of integer class label of shape (batch_size, ...). It must be the same
+            shape as the ``predictions`` tensor.
+        mask: ``torch.Tensor``, optional (default = None).
+            Currently must be ``None``.  We keep this parameter just to match the API.
+        """
+        if mask is not None:
```

Comment:
Yeah, it just wasn't obvious to me what the mask would do.  But you're right, it's not hard.  I'll put it in.